### PR TITLE
Add logging for service difference across clusters

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

For https://github.com/istio/istio/issues/46105

### CODE REVIEW

File: pilot/pkg/serviceregistry/aggregate/controller.go [hunk 1]
```diff
@@ -285,10 +285,14 @@ func (c *Controller) GetService(hostname host.Name) *model.Service {
...
 
+// mergeService only merges two clusters' k8s services
 func mergeService(dst, src *model.Service, srcRegistry serviceregistry.Instance) {
+	if !src.Ports.Equals(dst.Ports) {
+		log.Debugf("service %s defined from cluster %s is different from others", src.Hostname, srcRegistry.Cluster())
+	}
 	// Prefer the k8s HostVIPs where possible
 	clusterID := srcRegistry.Cluster()
-	if srcRegistry.Provider() == provider.Kubernetes || len(dst.ClusterVIPs.GetAddressesFor(clusterID)) == 0 {
+	if len(dst.ClusterVIPs.GetAddressesFor(clusterID)) == 0 {
```
Comment (ramaraochavali) [inline pilot/pkg/serviceregistry/aggregate/controller.go:295]: why this change?

Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/aggregate/controller.go:295]: This is a redudant check, the function merge services only from k8s registry.
I also wanted to remove `len(dst.ClusterVIPs.GetAddressesFor(clusterID)) == 0` check

**Review by ramaraochavali — APPROVED** [review]
